src/events/kafkajs.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Handlers for kafkajs instrumentation
+ */
+
+const uuid4 = require('uuid4');
+const tracer = require('../tracer.js');
+const eventInterface = require('../event.js');
+const moduleUtils = require('./module_utils.js');
+const { EPSAGON_HEADER } = require('../consts.js');
+
+
+/**
+ * acts as a middleware for `producer.send()`
+ * @param {object} messages the messages param to send
+ * @param {Kafka} producer producer object
+ * @returns {Promise} The response promise
+ */
+function kafkaMiddleware(messages, producer) {
+    let response = Promise.resolve();
+    try {
+        const { slsEvent: kafkaEvent, startTime } = eventInterface.initializeEvent(
+            'kafka',
+            messages.topic,
+            'send',
+            'kafkajs'
+        );
+
+        const epsagonId = uuid4();
+        // eslint-disable-next-line no-param-reassign
+        messages.messages = messages.messages.map((message) => {
+            if (!message.headers) {
+                // eslint-disable-next-line no-param-reassign
+                message.headers = {};
+            }
+            // eslint-disable-next-line no-param-reassign
+            message.headers[EPSAGON_HEADER] = epsagonId;
```

Comment:
you are setting the same id for all of the messages, this doesn't seem right

Reply:
You're right. it's on purpose at the moment because we don't support setting multiple IDs on the same event. Any other alternative might require nice amount of re-design, and might be overkill for this stage